x-pack/solutions/observability/plugins/observability_agent_builder/public/types.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { OnechatPluginStart } from '@kbn/onechat-plugin/public';
+import type { DiscoverSharedPublicStart } from '@kbn/discover-shared-plugin/public';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ObservabilityAgentBuilderPluginSetup {}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ObservabilityAgentBuilderPluginStart {}
```

Comment:
To differentiate from server start type

```suggestion
export interface ObservabilityAgentBuilderPluginPublicStart {}
```